src/spyglass/decoding/v1/sorted_spikes.py

```diff
@@ -0,0 +1,285 @@
+        position_info = pd.concat(position_info, axis=0).dropna()
+
+        # Get the waveform features for the selected units
+        merge_ids = (
+            (
+                SortedSpikesGroup.SortGroup
+                & {"sorted_spikes_group_name": key["sorted_spikes_group_name"]}
+            )
+        ).fetch("merge_id")
+
+        spike_times = [
+            SpikeSortingOutput.fetch_nwb({"merge_id": merge_id})[0][
+                "object_id"
+            ]["spike_times"].to_list()
+            for merge_id in merge_ids
+        ]
+        spike_times = list(chain(*spike_times))
+
+        # Get the encoding and decoding intervals
+        encoding_interval = (
+            IntervalList
+            & {
+                "nwb_file_name": key["nwb_file_name"],
+                "interval_list_name": key["encoding_interval"],
+            }
+        ).fetch1("valid_times")
+        is_training = np.zeros(len(position_info), dtype=bool)
+        for interval_start, interval_end in encoding_interval:
+            is_training[
+                np.logical_and(
+                    position_info.index >= interval_start,
+                    position_info.index <= interval_end,
+                )
+            ] = True
+        if "is_training" not in decoding_kwargs:
+            decoding_kwargs["is_training"] = is_training
+
+        decoding_interval = (
+            IntervalList
+            & {
+                "nwb_file_name": key["nwb_file_name"],
+                "interval_list_name": key["decoding_interval"],
+            }
+        ).fetch1("valid_times")
+
+        # Decode
+        classifier = SortedSpikesDetector(**decoding_params)
+
+        if key["estimate_decoding_params"]:
+            # if estimating parameters, then we need to treat times outside decoding interval as missing
+            # this means that times outside the decoding interval will not use the spiking data
+            # a better approach would be to treat the intervals as multiple sequences
+            # (see https://en.wikipedia.org/wiki/Baum%E2%80%93Welch_algorithm#Multiple_sequences)
+            is_missing = np.ones(len(position_info), dtype=bool)
+            for interval_start, interval_end in decoding_interval:
+                is_missing[
+                    np.logical_and(
+                        position_info.index >= interval_start,
+                        position_info.index <= interval_end,
+                    )
+                ] = False
+            if "is_missing" not in decoding_kwargs:
+                decoding_kwargs["is_missing"] = is_missing
+            results = classifier.estimate_parameters(
+                position_time=position_info.index.to_numpy(),
+                position=position_info[position_variable_names].to_numpy(),
+                spike_times=spike_times,
+                time=position_info.index.to_numpy(),
+                **decoding_kwargs,
+            )
+        else:
+            VALID_FIT_KWARGS = [
```

Comment:
It looks like this is in multiple scripts, such that is could be extracted elsewhere. Given the try/except on the imports, maybe v0 has old dependencies that spyglass no longer requires on the assumption that new installs will use v1? If so, that's a reason to prefer util over import from v0